components/NotificationOptions.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <component
+    :is="notificationType"
+    class="white text-center notification-list"
+    :class="{'mr-2' : smallScreen}"
+    :variant="smallScreen ? 'transparent' : ''"
+    toggle-class="notification-list__dropdown-toggle"
+    menu-class="notification-list__dropdown-menu"
+    lazy
+    right
+    aria-label="notifications"
+    @shown="loadLatestNotifications"
+  >
+    <template slot="button-content">
+      <div class="position-relative" :class="{'text-center small' : !smallScreen}">
+        <v-icon name="bell" scale="2" class="notification-list__icon" />
+        <b-badge v-if="unreadNotificationCount" variant="danger" class="notification-badge">
+          {{ unreadNotificationCount }}
+        </b-badge>
+        <div v-if="!smallScreen" class="nav-item__text">
+          Notifications
+        </div>
+      </div>
+    </template>
+    <b-dropdown-item class="text-right" link-class="notification-list__item">
+      <b-btn variant="white" size="sm" @click="markAllRead">
+        Mark all read
+      </b-btn>
+    </b-dropdown-item>
+    <b-dropdown-divider />
+    <b-dropdown-item v-for="notification in notifications" :key="'notification-' + notification.id" class="p-0 notpad" link-class="notification-list__item">
+      <Notification :notification="notification" class="p-0" @showModal="showAboutMe" />
+    </b-dropdown-item>
+    <infinite-loading :distance="distance" @infinite="loadMoreNotifications">
+      <span slot="no-results" />
+      <span slot="no-more" />
+      <span slot="spinner">
+        <b-img-lazy src="~/static/loader.gif" alt="Loading" />
+      </span>
+    </infinite-loading>
+  </component>
+</template>
+
+<script>
+const Notification = () => import('~/components/Notification')
+const InfiniteLoading = () => import('vue-infinite-loading')
+
+export default {
+  name: 'NotificationOptions',
+  components: {
+    InfiniteLoading,
+    Notification
+  },
+  props: {
+    distance: {
+      type: Number,
+      required: true
+    },
+    smallScreen: {
+      type: Boolean,
+      required: false,
+      default: false
+    }
+  },
+  computed: {
+    notificationType() {
+      return this.smallScreen ? 'b-dropdown' : 'b-nav-item-dropdown'
```

Comment:
Can you comment saying why we do this?

Reply:
Do you mean technically as in how the code is used or why we have two different styles of menu?

Reply:
Would something like this make sense?

// A different component needs to be created depending on the context in which it's used